### DISCUSSION TITLE
zh/guide/drivers/googledrive

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#29] [Announcements]
# zh/guide/drivers/googledrive

AList v3的文档

https://alist.nn.ci/zh/guide/drivers/googledrive.html

**Comment by kevzcr** [comment]
只剩下谷歌盘怎么也学不会加载，网上也找不到教程，太难了。。。不能说，全是泪，哗哗地。。。 程序员大大，在这给你作揖了。出个教程吧，其实谷歌云并不好用，但就想学习学习怎么加载，😰😰😰

**Reply by vpei** [reply]
我的谷歌云盘也不行，谷歌应该是需要代理的原因，还有一个迅雷云盘，有个验证码出不来，暂时没搞定。

**Reply by hzc54188** [reply]
我成功过，当时我把alist部署在服务器，快去试一下部署吧

**Reply by hzc54188** [reply]
可以v2 v3一起使用的，把v2端口必成5242，v2对迅雷云盘支持好一些

**Reply by Eight-God** [reply]
https://www.youtube.com/watch?v=Hktdyg7L0Bg 看这个教程，我挂成功了

**Reply by chinesecalvin8282** [reply]
状态: 

Post "https://www.googleapis.com/oauth2/v4/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

有办法解决吗？

**Reply by jiabingxian** [reply]
科学上网

**Comment by arthurfsy** [comment]
已成功通过“https://tool.nn.ci/google/request”获取client_id, client_secret, refresh_token，但是填入后点击‘提交’，显示以下错误，请问如何解决
Failed create storage in database: UNIQUE constraint failed: x_storages.mount_path

**Reply by HsinKwok** [reply]
这个报错，是数据库里面已经有了这个记录，改个挂载点就好了。不知为什么我这个存储列表里面要第二天才能看到挂载列表，才能修改。目前已挂好阿里，Gdrive，onedrive，中国移动云盘。

**Comment by hzc54188** [comment]
https://www.googleapis.com/oauth2/v4/token": read tcp 192.168.1.50:3050->172.217.160.106:443: wsarecv: An existing connection was forcibly closed by the remote host.这是什么意思

**Comment by Sostay** [comment]
Failed init storage: Post "https://www.googleapis.com/oauth2/v4/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
一直提示这个……

**Reply by 95121808** [reply]
我也是这个问题，你解决了吗？

**Reply by iYaoyaoo** [reply]
我也是，请问有解决吗

**Reply by chinesecalvin8282** [reply]
同一个问题，现在有大大解决了吗？

**Reply by hemingwang0902** [reply]
这个超时的原因是因为需要科学上网才能正常访问到，使用 Clash 科学上网，同时开启 系统代理 和 TUN 模式即可

**Reply by AdamQwy** [reply]
一定要《Clash 配置 - 开启 TUN 转发》

**Comment by DuangXT** [comment]
心好累，完全不知道为啥

无法登录，因为“rclone”发送的请求无效。您可稍后再试，也可向开发者求助。 详细了解此错误
如果您是“rclone”的开发者，请参阅错误详情。
错误 400： invalid_request

**Reply by Harrison3-1122** [reply]
我也是

**Reply by DuangXT** [reply]
其实可以隔天试试，应该是冲烂了，谷歌好像是有每天配额的亚子

**Reply by KimmyLueng** [reply]
自己动手，在Google Cloud自建一个吧。反正是自用

**Reply by simpread** [reply]
看看 OAuth 同意屏幕-发布状态是不是正式版，调整为测试版，测试用户添加自己的谷歌邮箱

**Comment by bloding654** [comment]
太难了 在境外用国内的盘速度很慢 不知道是不是网络限制的原因还是其它什么原因。。想用谷歌和pikpak 根目录又获取不了 ~~

**Comment by yanygm** [comment]
每天refresh_token都会过期，要重新获取

**Reply by Muggles001** [reply]
解决了吗

**Comment by 95121808** [comment]
这谷歌网盘真实太难了，按照教程走最后给我来个超时，各国节点都是超时，可怎么办才好？

**Comment by azengjing** [comment]
登录发送的请求无效，反复多次无果，只好放弃Google云盘。只要能加载上天翼云盘和阿里云盘已经onedrive就满意了，icloud也经常用，但使用webdev也是死活加载不上。

**Comment by KimmyLueng** [comment]
谷歌相册能读到 共享影集、全部媒体、全部影集 这三个文件夹，但不能显示照片。有人知道解决方法吗？

**Comment by HsinKwok** [comment]
https://tool.nn.ci/google/callback 这是回调地址，参考onedrive得出来的。。。

**Comment by simpread** [comment]
按照这个视频https://www.youtube.com/watch?v=Hktdyg7L0Bg成功了。之前出现
“自己无法登录，因为“rclone”发送的请求无效。您可稍后再试，也可向开发者求助。 详细了解此错误
如果您是“rclone”的开发者，请参阅错误详情。
错误 400： invalid_request”的错误。
解决：发现OAuth 同意屏幕-发布状态-为正式版而不是测试版，调整过后就成了

**Reply by simpread** [reply]
怕没说清楚，发布状态用测试版

**Reply by Muggles001** [reply]
但是测试版7天一过期怎么办

**Comment by 10YuJi** [comment]
Failed init storage but storage is already created: google_drive.Addition.ChunkSize: readUint64: unexpected character: �, error found in #10 byte of ...|nk_size":"5"}|..., bigger context ...|OCSPX-3bejJ3lEAu7vsqBi3Aou45opxd8O","chunk_size":"5"}|...一直出现这个报错

**Reply by 10YuJi** [reply]
应该是3.4的问题

**Comment by 95121808** [comment]
用的是测试版，自己的邮箱。我看留言板有说用正式版，其实我都试过，看来还要再试试。
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月22日(星期二) 晚上10:15
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
看看 OAuth 同意屏幕-发布状态是不是正式版，调整为测试版，测试用户添加自己的谷歌邮箱

**Reply by HsinKwok** [reply]
你这是什么平台发送的留言？电话露了

**Comment by 95121808** [comment]
准备试一下再改成正式版
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月22日(星期二) 晚上10:13
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
按照这个视频https://www.youtube.com/watch?v=Hktdyg7L0Bg成功了。之前出现
 “自己无法登录，因为“rclone”发送的请求无效。您可稍后再试，也可向开发者求助。 详细了解此错误
 如果您是“rclone”的开发者，请参阅错误详情。
 错误 400： invalid_request”的错误。
 解决：发现OAuth 同意屏幕-发布状态-为正式版而不是测试版，调整过后就成了

**Reply by Muggles001** [reply]
成功没

**Comment by 95121808** [comment]
问题还未解决，会参考各位大佬的建议。
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月22日(星期二) 晚上11:46
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
应该是3.4的问题

**Reply by HsinKwok** [reply]
重点是Google driver api回调地址要https://tool.nn.ci/google/callback 这是回调地址，参考onedrive得出来的。。。

**Comment by 95121808** [comment]
但是Alist的GC好像没有这个可以填写的位置。
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月23日(星期三) 凌晨1:05
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
重点是Google driver api回调地址要https://tool.nn.ci/google/callback 这是回调地址，参考onedrive得出来的。。。

**Comment by 95121808** [comment]
号码已作废，感谢关心，QQ邮件回复。
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年11月23日(星期三) 凌晨1:06
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
你这是什么平台发送的留言？电话露了

**Comment by 95121808** [comment]
好嘞，明天就装上测试。



---Original---
From: ***@***.***&gt;
Date: Fri, Nov 25, 2022 00:07 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
怕没说清楚，发布状态用测试版

**Comment by 95121808** [comment]
可我现在就是测试版啊😂



---Original---
From: ***@***.***&gt;
Date: Fri, Nov 25, 2022 00:07 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
怕没说清楚，发布状态用测试版

**Comment by 95121808** [comment]
Failed init storage: Post "https://www.googleapis.com/oauth2/v4/token": read tcp ***********8:50560->************34:443: read: connection reset by peer

**Reply by 95121808** [reply]
各位大佬，总是提示初始化失败，请问还有救吗？

**Reply by catcafeman** [reply]
你好，问一下你的code获取成功了吗，我的会这样
禁止访问：“rclone”的请求无效
*********@**********.com
无法登录，因为“rclone”发送的请求无效。您可稍后再试，也可向开发者求助。 详细了解此错误
如果您是“rclone”的开发者，请参阅错误详情。
错误 400： invalid_request

**Comment by catcafeman** [comment]
failed get objs: failed to list objs: Get "https://www.googleapis.com/drive/v3/files?fields=files%28id%2Cname%2CmimeType%2Csize%2CmodifiedTime%2CthumbnailLink%2CshortcutDetails%29%2CnextPageToken&includeItemsFromAllDrives=true&orderBy=folder%2Cname%2CmodifiedTime+desc&pageSize=1000&pageToken=&q=%27root%27+in+parents+and+trashed+%3D+false&supportsAllDrives=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

求助

**Reply by simpread** [reply]
按照这个视频https://www.youtube.com/watch?v=Hktdyg7L0Bg
跟一遍试试

**Reply by kaiser20106** [reply]
我就是按这个视频来的，还是报错，就是这个错误

**Comment by 95121808** [comment]
谢谢，前段时间自我隔离，操作好马上回复，感谢大佬



---Original---
From: ***@***.***&gt;
Date: Mon, Dec 5, 2022 10:11 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
按照这个视频https://www.youtube.com/watch?v=Hktdyg7L0Bg
 跟一遍试试

**Comment by 95121808** [comment]
隔离，还未回家操作



---Original---
From: ***@***.***&gt;
Date: Sat, Dec 3, 2022 15:58 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
我也是，请问有解决吗

**Comment by rongzhijie7** [comment]
我试了很多次，在自己的docker上怎么也过不去，后来在甲骨文云上试了一下，就可以用了。
推测还是网络无法直连谷歌网络的问题，但是我已经用CLASH把本地网络改成全局代理了，我本地访问谷歌也没问题（有问题的话也拿不到ID和密钥了），可一直提示错误，把我搞得也很懵逼。

**Comment by 95121808** [comment]
应该是网络问题，我找个甲骨文的代理试试吧咱俩一样情况
&nbsp;


15977725906
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年12月14日(星期三) 下午2:12
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
我试了很多次，在自己的docker上怎么也过不去，后来在甲骨文云上试了一下，就可以用了。
 推测还是网络无法直连谷歌网络的问题，但是我已经用CLASH把本地网络改成全局代理了，我本地访问谷歌也没问题（有问题的话也拿不到ID和密钥了），可一直提示错误，把我搞得也很懵逼。

**Comment by 95121808** [comment]
无语，我盘没了😂



---Original---
From: ***@***.***&gt;
Date: Wed, Dec 21, 2022 23:18 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
https://www.youtube.com/watch?v=Hktdyg7L0Bg 看这个教程，我挂成功了

**Comment by kugimiya530** [comment]
不好意思 请问谷歌云盘可以直接挂载  云盘的"根目录"吗

因为好像都只能挂一级目录  因为根目录没有资料夹id...

**Comment by fengyupiaoyao75** [comment]
failed get objs: failed to list objs: Get "https://www.googleapis.com/drive/v3/files?fields=files%28id%2Cname%2CmimeType%2Csize%2CmodifiedTime%2CthumbnailLink%2CshortcutDetails%29%2CnextPageToken&includeItemsFromAllDrives=true&orderBy=folder%2Cname%2CmodifiedTime+desc&pageSize=1000&pageToken=&q=%27root%27+in+parents+and+trashed+%3D+false&supportsAllDrives=true": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

超时了

**Comment by 95121808** [comment]
我网盘被折腾的收回了



---Original---
From: ***@***.***&gt;
Date: Wed, Feb 1, 2023 00:17 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
解决了吗

**Comment by chinesecalvin8282** [comment]
2023年2月安装，状态: 

Post "https://www.googleapis.com/oauth2/v4/token": context deadline exceeded (Client.Timeout exceeded while awaiting headers)

有办法解决吗？

**Comment by ykc1043** [comment]
挂载后在网页端可以播放视频但不能调整进度条，webdav挂载只能下载不能播放要怎么办

**Comment by gradendripar** [comment]
只能获取我的云端硬盘吗 共享云端硬盘没法获取吗

**Comment by BurtSweet** [comment]
User rate limit exceeded.: [{usageLimits userRateLimitExceeded User rate limit exceeded. }]
挂上了,能预览.上传就报错.....网上搜了说是谁么配额问题怎么解啊

**Reply by Muggles001** [reply]
现贼还能在线播放吗

**Reply by BurtSweet** [reply]
可以，现在谷歌云盘最大的问题是token有效期太短了，几天就失效了。

**Comment by Muggles001** [comment]
加我q1211220192



---Original---
From: ***@***.***&gt;
Date: Tue, Apr 11, 2023 10:59 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
User rate limit exceeded.: [{usageLimits userRateLimitExceeded User rate limit exceeded. }]
 挂上了,能预览.上传就报错.....网上搜了说是谁么配额问题怎么解啊

**Comment by hemingwang0902** [comment]
如果提示 Client.Timeout exceeded while awaiting headers，使用 Clash 科学上网，同时开启 系统代理（System Proxy） 和 TUN 模式（TUN Mode）即可。注意：开启 TUN 模式之前请确保安装了“服务模式”（Service Mode），即该项旁边的地球小图标显示为绿色。

**Comment by S001afe** [comment]
“nn.ci”尚未完成 Google 验证流程。此应用正在测试中，仅供已获开发者批准的测试人员使用。如果您认为自己应有权使用，请联系开发者。
如果您是“nn.ci”的开发者，请参阅错误详情。
错误 403： access_denied

**Reply by jing8263xiao** [reply]
我也是，解决没

**Reply by OdinShiva** [reply]
我也是+1

**Comment by Muggles001** [comment]
只能下载，在线播放失败咋整，跟浏览器无关

**Reply by hemingwang0902** [reply]
应该是你的代理服务器网速问题，我访问的时候也是加载图片有点慢...

**Reply by Muggles001** [reply]
不行，我切换了大带宽服务器依旧无法播放

**Reply by hemingwang0902** [reply]
我的能正常打开图片，由于我的相册中没有视频，所以视频能不能正常播放就无法验证啦。

**Comment by Muggles001** [comment]
你能正常播放视频吗



---Original---
From: ***@***.***&gt;
Date: Fri, May 12, 2023 11:31 AM
To: ***@***.***&gt;;
Cc: "Clifford ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
应该是你的代理服务器网速问题，我访问的时候也是加载图片有点慢...

**Comment by kiddyyang** [comment]
按教程都搞定了，最后一步保存存储时候一直报这个错，开了全局代理都没用
Failed init storage: Post "https://www.googleapis.com/oauth2/v4/token": net/http: TLS handshake timeout

**Comment by KimmyLueng** [comment]
我把我自己的方法介绍一下吧
#安装rclone在本地主机上，然后命令行（终端）输入rclone config；
#新建并添加Google Drive
##client_id>     #回车
##client_secret>      #回车
##Scope that rclone should use when requesting access from drive.
Choose a number from below, or type in your own value.
Press Enter to leave empty.
 1 / Full access all files, excluding Application Data Folder.
   \ (drive)
 2 / Read-only access to file metadata and file contents.
   \ (drive.readonly)
   / Access to files created by rclone only.
 3 | These are visible in the drive website.
   | File authorization is revoked when the user deauthorizes the app.
   \ (drive.file)
   / Allows read and write access to the Application Data folder.
 4 | This is not visible in the drive website.
   \ (drive.appfolder)
   / Allows read-only access to file metadata but
 5 | does not allow any access to read or download file content.
   \ (drive.metadata.readonly)
scope>                    #选1
##service_account_file>   #回车
##Edit advanced config?   #选No (default)
##Use auto config?    #选Yes (default)，这时会跳到谷歌云盘的授权网页
##选择要授权的账号，在跳转的网页里点允许，网页跳到Success!的页面
##然后返回命令行（终端），这时提示Configure this as a Shared Drive (Team Drive)?       #选No (default)
##这时命令行（终端）里跳出
  Configuration complete.
Options:
- type: drive
- scope: drive
- token: {"access_token":"***************************","token_type":"Bearer","refresh_token":"********************************************","expiry":"****-**-**T**:**:**.483273+08:00"}
- team_drive: 
Keep this "**" remote?
y) Yes this is OK (default)
e) Edit this remote
d) Delete this remote
y/e/d>                          #上面的refresh_token就是后台上的刷新令牌，复制令牌并输入d把上面的remote删除
#最后点击    https://alist.nn.ci/tool/google/request    ，点选Use Rclone client，复制client_id和client_secret到后台，保存，添加成功
#以上是利用rclone的OAuth 客户端来绑定谷歌云盘，因为这个是rclone正式发布的应用，目前未遇到7天失效的问题

**Reply by KimmyLueng** [reply]
简版：
第一步本地rclone添加谷歌云盘，取最后结果里的refres_token；
第二步在   https://alist.nn.ci/tool/google/request   里获取rclone的id和secret；
第三步把上面三个值填入后台对应位置，保存

**Comment by Muggles001** [comment]
你这真麻烦啊



---Original---
From: ***@***.***&gt;
Date: Sat, Jun 24, 2023 17:17 PM
To: ***@***.***&gt;;
Cc: "Clifford ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
我把我自己的方法介绍一下吧
 #安装rclone在本地主机上，然后命令行（终端）输入rclone config；
 #新建并添加Google Drive
 ##client_id&gt;     #回车
 ##client_secret&gt;      #回车
 ##Scope that rclone should use when requesting access from drive.
 Choose a number from below, or type in your own value.
 Press Enter to leave empty.
 1 / Full access all files, excluding Application Data Folder.
 \ (drive)
 2 / Read-only access to file metadata and file contents.
 \ (drive.readonly)
 / Access to files created by rclone only.
 3 | These are visible in the drive website.
 | File authorization is revoked when the user deauthorizes the app.
 \ (drive.file)
 / Allows read and write access to the Application Data folder.
 4 | This is not visible in the drive website.
 \ (drive.appfolder)
 / Allows read-only access to file metadata but
 5 | does not allow any access to read or download file content.
 \ (drive.metadata.readonly)
 scope&gt;                    #选1
 #service_account_file&gt;   #回车
 #Edit advanced config?   #选No (default)
 #Use auto config?    #选Yes (default)，这时会跳到谷歌云盘的授权网页
 #选择要授权的账号，在跳转的网页里点允许，网页跳到Success!的页面
 #然后返回命令行（终端），这时提示Configure this as a Shared Drive (Team Drive)?       #选No (default)
 #这时命令行（终端）里跳出
 Configuration complete.
 Options:
  
type: drive
 
scope: drive
 
token: {"access_token":"","token_type":"Bearer","refresh_token":"***********","expiry":"****--T::.483273+08:00"}
 
team_drive:
 Keep this "**" remote?
 y) Yes this is OK (default)
 e) Edit this remote
 d) Delete this remote
 y/e/d&gt;                          #上面的refresh_token就是后台上的刷新令牌，复制令牌并输入d把上面的remote删除
 #最后点击https://alist.nn.ci/tool/google/request，点选Use Rclone client，复制client_id和client_secret到后台，保存，添加成功
 #以上是利用rclone的OAuth 客户端来绑定谷歌云盘，因为这个是rclone正式发布的应用，目前未遇到7天失效的问题

**Reply by KimmyLueng** [reply]
总而言之，言而总之。第一步本地rclone添加谷歌云盘，取最后结果里的refres_token；第二步在 https://alist.nn.ci/tool/google/request 里获取rclone的id和secret；第三步把上面三个值填入后台对应位置，保存

**Comment by BurtSweet** [comment]
确实过几天token就过期了啊，要不开发大大们发布正式版？

**Comment by 95121808** [comment]
我无限网盘没了，有带着玩的大佬吗？&nbsp;


Docker-D
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年8月28日(星期一) 晚上8:12
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
确实过几天token就过期了啊，要不开发大大们发布正式版？

**Comment by Muggles001** [comment]
挂载一年了没掉过



---Original---
From: ***@***.***&gt;
Date: Tue, Aug 29, 2023 23:18 PM
To: ***@***.***&gt;;
Cc: "Clifford ***@***.******@***.***&gt;;
Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)




 
可以，现在谷歌云盘最大的问题是token有效期太短了，几天就失效了。

**Comment by 95121808** [comment]
有没有大佬能给一个完美的教程&nbsp;


Docker-D
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Clifford ***@***.***&gt;; 
发送时间: 2023年8月29日(星期二) 晚上11:44
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 挂载一年了没掉过 
 
 
 
 ---Original--- 
 From: ***@***.***&amp;gt; 
 Date: Tue, Aug 29, 2023 23:18 PM 
 To: ***@***.***&amp;gt;; 
 Cc: "Clifford ***@***.******@***.***&amp;gt;; 
 Subject: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29) 
 
 
 
 
   
 可以，现在谷歌云盘最大的问题是token有效期太短了，几天就失效了。

**Reply by KimmyLueng** [reply]
看我上面的，用rclone生成的refres_token和网页中间那个rclone的内容
![image](https://github.com/alist-org/docs/assets/54270610/d4e8f294-b5b7-421f-9fb9-388b9243ee5c)

**Comment by 95121808** [comment]
感谢大佬，我试试，成功了再回帖。&nbsp;


Docker-D
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年9月11日(星期一) 晚上8:10
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [alist-org/docs] zh/guide/drivers/googledrive (Discussion #29)





 
看我上面的，用rclone生成的refres_token和网页中间那个rclone的内容

**Comment by nanqic** [comment]
请问可以挂载别人共享给我的文件夹吗？
一般情况 7 天是不是要重新获取一次 token？有长久一点的办法吗

**Comment by healson** [comment]
最后一步申请授权时，google返回400错误，按照下面兄弟提供的方法将OAuth从测试版改为正式版还是不行，后来仔细看是重定向问题，于是将上面的重定向地址https://alist.nn.ci/tool/google/callback填到了OAuth的信任重定向页面里就好了。

**Comment by deluxekuma** [comment]
登陆Google账号时提示这个怎么办啊😰

显示如下

已封鎖存取權：「nn.ci」未完成 Google 驗證程序

「nn.ci」尚未完成 Google 驗證程序。這個應用程式目前處於測試階段，只有獲得開發人員核准的測試人員可以存取。如果您認為自己應該擁有存取權，請與開發人員聯絡。
如果您是「nn.ci」的開發人員，請參閱錯誤詳細資料。
錯誤代碼 403： access_denied
